### ISSUE TITLE
Duplicate barcodes not rendered

### ISSUE THREAD

**Issue by peekxc** [rrrlw/ggtda#3]
I think there's an issue with the barcode geom when two given bars are not unique. I have 28 barcodes, however 3 of them have exactly the same birth/death times, so only 25 get rendered. 

Here's an example with the embedded toy data I was working with:

``` r
## Generated data 
dim <- c(0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,1)
birth <- c(0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0.213226,0.213225)
death <- c(0.00718,0.008658,0.00866,0.033702,0.033704,0.041683,0.041683,0.049228,0.049229,0.103626,0.103626,0.108313,0.108314,0.116322,0.116323,0.122399,0.1224,0.145653,0.145654,0.169471,0.169472,0.207892,0.207892,0.208636,0.208637,Inf,Inf,Inf)
bar <- data.frame(birth=birth, death=death, dim=dim)

## Plot barcodes w/ ggtda
library("ggtda")
#> Loading required package: ggplot2
#> Warning: package 'ggplot2' was built under R version 3.5.2
barcodes <- data.frame(appear=bar$birth, disappear=bar$death, dim=as.character(bar$dim))
p1 <- ggplot(barcodes, aes(start = appear, end = disappear, colour = dim, shape = dim)) +
      geom_barcode() +
      theme_tda()

## Working linerange w/ id_ dimension
barcodes_id <- cbind(id_ = seq(length(bar$birth)), barcodes)
p2 <- ggplot(barcodes_id) + geom_linerange(aes(x = id_, ymin = appear, ymax=disappear, colour = dim))
```

<sup>Created on 2019-05-16 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

If you inspect p1 vs p2, you'll see that p1 plots 25 bars while all 28 get plotted w/ p2.

**Comment by corybrunson** [comment]
@peekxc thanks for raising this issue!

Below is a reduced reproducible example. Notice that, as one might expect, bars from different groups (e.g. dimensions) are _not_ de-duplicated.

One of us should be able to look into this in the next few days.

``` r
library(ggtda)
#> Loading required package: ggplot2
data1 <- data.frame(
  birth = c(0,0,0,1),
  death = c(1,2,2,2),
  dim = as.character(c(0,0,0,0))
)
ggplot(data1, aes(start = birth, end = death, color = dim)) +
  geom_barcode() +
  theme_tda()
```

![](https://i.imgur.com/pDGuyBP.png)

``` r
data2 <- data.frame(
  birth = c(0,0,0,1),
  death = c(1,2,2,2),
  dim = as.character(c(0,0,1,1))
)
ggplot(data2, aes(start = birth, end = death, color = dim)) +
  geom_barcode() +
  theme_tda()
```

![](https://i.imgur.com/6QvMsq9.png)

<sup>Created on 2019-05-16 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

**Comment by peekxc** [comment]
@corybrunson thanks for the update, and kudos for the much simpler example. Yes the first example demonstrates what I'm running into. No rush, and thanks for the nice package

**Comment by corybrunson** [comment]
@peekxc could you verify that the fix at 236b796 (branch `dupes`) resolves this issue on your original data?

**Comment by peekxc** [comment]
Indeed, it does! Thanks for the fix!
